internal/controller/api/v1alpha1/image_types.go

```diff
@@ -17,25 +17,28 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // ImageSpec defines the desired state of Image
 type ImageSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of Image. Edit Image_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// The name of the Cluster CR
+	ClusterName  *corev1.ObjectReference `json:"cluster_name,omitempty"`
+	SSHPublicKey string                  `json:"ssh_public_key,omitempty"`
+	// The name of the secret containing the pull secret
+	PullSecretName        string `json:"pull_secret_name,omitempty"`
+	IgnitionOverrides     string `json:"ignition_overrides,omitempty"`
+	StaticIpConfiguration string `json:"static_ip_configuration,omitempty"`
 }
 
 // ImageStatus defines the observed state of Image
 type ImageStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	State               string       `json:"state,omitempty"`
+	SizeBytes           int          `json:"size_bytes,omitempty"`
+	DownloadUrl         int          `json:"download_url,omitempty"`
+	CreationTimestamp   *metav1.Time `json:"creation_timestamp,omitempty"`
```

Comment:
maybe it should be the upload time instead? @avishayt what do you think?

Reply:
I named it `CreationTimestamp` based on [this comment](https://github.com/openshift/enhancements/pull/574/files#r550045709) from @avishayt, I can change it if this is what we want.